File: python/largest-series-product/largest_series_product.py
from functools import reduce
def accumulate_plus(nums):
    return reduce(lambda x, y: x * int(y), nums, 1)

def largest_product(series, size):
    if size < 0:
        raise ValueError('Invalid size')
    whole_series = [series[i:i+size] for i in range(len(series)-size+1)]
    return max(map(accumulate_plus, whole_series))

File: ruby/largest-series-product/largest_series_product.rb
class Series
  def initialize(str)
    @str = str
  end

  def largest_product(num)
    return 1 if num == 0
    raise ArgumentError if num < 0 || num > @str.size || @str.empty? || !@str.scan(/\D/).empty?
    @str.chars.map { |e| e.to_i }.each_cons(num).map { |e| e.reduce(:*) }.max
  end

end

module BookKeeping
  VERSION = 3
end